Added supported environments section for RC documentation (#21107)
DataDog/documentation: content/en/agent/remote_config/_index.md

  | NOT CONNECTED       | The Agent cannot be found in the Remote Configuration service and could have `remote_config.enabled` set to true or false in its `datadog.yaml` configuration file. Check your local Agent configuration or your proxy settings. This status displays when the Agent version is higher than `7.41.1` but lower than `7.45.0`.            | 
  | UNSUPPORTED AGENT   | The Agent is on a version that is not Remote Configuration capable. To fix this issue, update the Agent to the latest available version. |

The following table describes the meaning of each Tracing library status:

  | Tracing library Status| Description                                      |
  |------------------|--------------------------------------------------|
  | CONNECTED      | The Tracing library is successfully connected to the Remote Configuration service through the associated Agent. This is the optimal state you want your Tracing library to be in for Remote Configuration.                                               |    

>   | UNAUTHORIZED          | The Tracing library is associated with an Agent which doesn’t have “Remote Config Read” permission on its API key. To fix the issue, you need to enable Remote Configuration capability on the API Key used by the Agent associated with the Tracing library.| 

Reviewer: ```suggestion
  | UNAUTHORIZED          | The Tracing library is associated with an Agent which doesn't have “Remote Config Read” permission on its API key. To fix the issue, you need to enable Remote Configuration capability on the API Key used by the Agent associated with the Tracing library.| 
```

>   | UNAUTHORIZED          | The Tracing library is associated with an Agent which doesn't have “Remote Config Read” permission on its API key. To fix the issue, you need to enable Remote Configuration capability on the API Key used by the Agent associated with the Tracing library.| 

Reviewer: ```suggestion
  | UNAUTHORIZED          | The Tracing library is associated with an Agent which doesn't have `Remote Config Read` permission on its API key. To fix the issue, you need to enable Remote Configuration capability on the API Key used by the Agent associated with the Tracing library.| 
```

  | CONNECTION ERROR        |   The Tracing library deployed in your environment is associated with an Agent that has remote_config.enabled set to true in its `datadog.yaml` configuration file, however, the agent cannot be found in the Remote Configuration service. The most likely cause of this is that the associated Agent is unable to reach Remote Configuration [endpoints][17]. To fix the issue, you need to allow outbound HTTPS access to Remote Configuration endpoints from your environment.  

>   | DISABLED       |   The Tracing library deployed in your environment is associated with an Agent that has `remote_config.enabled` set to false in its `datadog.yaml` configuration file. This could be set deliberately or mistakenly. You will want to set `remote_config.enabled` to true if you want to enable Remote Configuration on the associated Agent.  | 

Reviewer: ```suggestion
  | DISABLED       |   The Tracing library deployed in your environment is associated with an Agent that has `remote_config.enabled` set to false in its `datadog.yaml` configuration file. This could be set deliberately or mistakenly. To enable Remote Configuration on the associated Agent, set `remote_config.enabled` to true.  | 
```

  | NOT CONNECTED       | The Tracing library cannot be found in the Remote Configuration service and is associated with an Agent that could have `remote_config.enabled` set to true or false in its `datadog.yaml` configuration file. Check your local Agent configuration or your proxy settings.| 
  | UNSUPPORTED AGENT   | The Tracing library is associated with an Agent which is not Remote Configuration capable. To fix this issue, update the associated Agent software to the latest available version. |
  | NOT DETECTED   | The Tracing library does not support Remote Configuration. To fix this issue, update the Tracing library software to the latest available version. |

>   | UNKNOWN   | The Tracing library status is unknown as we cannot determine if an Agent is associated with the Tracing library. This could be because the Agent is deployed on a fully managed serverless container service such as AWS Fargate or due to other reasons. |

Reviewer: ```suggestion
  | UNKNOWN   | The Tracing library status is unknown, and it can't be determined if an Agent is associated with the Tracing library. For example, this could be because the Agent is deployed on a fully managed serverless container service like AWS Fargate. |
```

{{% /tab %}}
{{< /tabs >}}

## Supported environments

> Remote Configuration works in environments where Datadog Agent is deployed. For a Serverless Container service such as AWS Fargate, you will not see the underlying hosts in the Remote Configuration onboarding workflow. Remote Configuration is not supported for Serverless Container Managed Apps (AWS App Runner, Azure Container Apps, Google Cloud Run) and Functions deployed via Container Packaging (AWS Lambda, Azure Functions, Google Cloud Functions)

Reviewer: ```suggestion
Remote Configuration works in environments where the Datadog Agent is deployed. For a Serverless Container service like AWS Fargate, the underlying hosts do not appear in the Remote Configuration onboarding workflow. Remote Configuration does not support Serverless Container Managed Apps (AWS App Runner, Azure Container Apps, Google Cloud Run) and Functions deployed via Container Packaging (AWS Lambda, Azure Functions, Google Cloud Functions)
```
Slight rewording to make the Vale checks happy.

{{% /tab %}}
{{< /tabs >}}

## Supported environments

> Remote Configuration works in environments where the Datadog Agent is deployed. For a Serverless Container service like AWS Fargate, the underlying hosts do not appear in the Remote Configuration onboarding workflow. Remote Configuration does not support Serverless Container Managed Apps (AWS App Runner, Azure Container Apps, Google Cloud Run) and Functions deployed via Container Packaging (AWS Lambda, Azure Functions, Google Cloud Functions)

Reviewer: ```suggestion
Remote Configuration works in environments where the Datadog Agent is deployed. For a Serverless Container service like AWS Fargate, the underlying hosts do not appear in the Remote Configuration onboarding workflow. Remote Configuration does not support Serverless Container Managed Apps (AWS App Runner, Azure Container Apps, Google Cloud Run) and Functions deployed with Container Packaging (AWS Lambda, Azure Functions, Google Cloud Functions)
```
